src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -379,7 +379,9 @@ private static bool IsFloatingPointExpressionOfUnknownPrecision(BoundExpression
                     {
                         var mg = (BoundMethodGroup)rewrittenOperand;
                         Debug.Assert(oldNodeOpt.SymbolOpt is { });
-                        return new BoundFunctionPointerLoad(oldNodeOpt.Syntax, oldNodeOpt.SymbolOpt, type: funcPtrType, hasErrors: false);
+                        return new BoundFunctionPointerLoad(oldNodeOpt.Syntax, oldNodeOpt.SymbolOpt,
+                                                            constrainedToTypeOpt: oldNodeOpt.SymbolOpt.IsStatic && oldNodeOpt.SymbolOpt.IsAbstract ? mg.ReceiverOpt?.Type : null,
```

Comment:
>? [](start = 163, length = 1)

Should we just dereference this, or give some kind of an error to handle the case? If we have a static abstract, we need to have a type or further layers are going to crash, right?

Reply:
> Should we just dereference this, or give some kind of an error to handle the case? If we have a static abstract, we need to have a type or further layers are going to crash, right?

I prefer to not surface this as a NullReference exception

Reply:
An exception in the emit layer is sufficient to deal with this unexpected situation

Reply:
> I prefer to not surface this as a NullReference exception

That's fine, but we're seriously busted at this point, right? Should we consider throwing an unreachable or something?

---
In reply to: [616239428](https://github.com/dotnet/roslyn/pull/52685#discussion_r616239428) [](ancestors = 616239428)

Reply:
> That's fine, but we're seriously busted at this point, right? Should we consider throwing an unreachable or something?

An emit layer takes care of this

Reply:
My intent is to not perform any validation here. Null reference is not the only thing we should worry about. The code matches the intent.